### DISCUSSION TITLE
How to rename only a folder with supabase js client

### DISCUSSION THREAD

**Discussion by haexhub** [supabase/supabase#19418] [Questions] [answered]
Hi supabase team :)

I would like to know, if it is possible to only rename a folder in supabase with the supabase js client, without finding and touching every file inside that folder.
For example here:
I want to rename folder parent to super_parent

![Bildschirmfoto vom 2023-12-01 15-22-21](https://github.com/supabase/supabase/assets/29780781/aeb3619b-49fd-4b04-ba61-f02b84c67535)

There exists the move function in supabase js client, but there you only can rename/move files not folders. This doesn't work:

    await supabase.storage.from("bucket_id").move("parent", "super_parent")

If you want to rename just a folder, you have to find all child files, child directories and there files recursive and rename every single file. Thats quite a lot of effort.

So I would like to know, if there is a shorthand version for this?

**Comment by GaryAustin1** [comment, marked as answer]
No you have to rename each file.  There is not a folder structure in the storage-api.  That is just a concept for RLS using string processing of the path name.

**Reply by haexhub** [reply]
Thanks for the quick response!
I know that there are no real folders. But I was hoping that there is a more convenient way of renaming paths. This way is quite incomprehensible, especially when you have deeply nested folder structures with many files down the hierarchy. 
Are you planning to implement such a function by any chance? I mean it would be sufficient, if you could just use the move function for this. In the dashboard it is quite easy to rename a folder name. How do you handle this under the hood?

**Reply by GaryAustin1** [reply]
I am not an employee.  I've not seen any code in the supabase/storage github along those lines.  The source code for studio is in supabase/supabase so it is possible to see how they do that in studio, but it will be done 1 file at a time as there is no API call to do and you can't just change the names in the storage.objects table so have to make storage API calls.
